api: Fix vm.add-device argument type

fidencio opened the pull request:
The add_device() function, from the device manager code, takes a DeviceConfig as a parameter, instead of a VmAddDevice.

Signed-off-by: Fabiano Fidêncio <fabiano.fidencio@intel.com>
Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>

likebreath reviewed (commented):
Good catch! Please also remove the `VmAddDevice` from the yaml file as it was replaced by `DeviceConfig` (for a long time).

likebreath commented:
FWIW, the reason why it happened to be working fine on the client side (e.g. Kata ) is that the `VmAddDevice` and `DeviceConfig ` share most of the fields besides `pci_segment` (which is optional for serilization and not being used by the client.

fidencio commented:
> Good catch! Please also remove the `VmAddDevice` from the yaml file as it was replaced by `DeviceConfig` (for a long time).

Changed, and also updated the commit message with the info you provided.
